### ISSUE TITLE
Proposal: Develop a Junit4 Engine

### ISSUE THREAD

**Issue by Drakojin** [testIT-LivingDoc/livingdoc2#110]
To support the still widely used junit 4.12 we should develop an engine which entails a junit4 Runner and/or JunitRule.

If we are not inclined to do that we should at least consider how we manage to convince people to switch to junit5 ;)

**Comment by slu-it** [comment]
**Migration to JUnit Platform:**
- remove `junit:junit` dependency from test classpath
- add `org.junit.vintage:junit-vintage-engine` to test classpath
- update build tool:
  - **Gradle:** use Gradle >= `v4.6`
  - **Maven:** use `maven-surefire-plugin` >= `v2.22.0`

*Using LivingDoc with JUnit Platform:*
- add `org.livingdoc:livingdoc-api` to test classpath
- add `org.livingdoc:livingdoc-junit-engine` to test runtime classpath

**Comment by slu-it** [comment]
Most people are already using JUnit 5 / Platform without even knowing it. If you are using IntelliJ >= 2017.1 - you are actually using the vintage engine to execute your tests in the IDE.

**Comment by Drakojin** [comment]
"most" people don't use intellij >= 2017.1 ;)

**Comment by slu-it** [comment]
Yeah, but i think it's the same for eclipse as well. If you are using current versions of your IDE, you're using JUnit Platform without realising it - that's my point.

**Comment by slu-it** [comment]
I'm not against writing a small module for JUnit 4 support - but it should start and stop at the MVP level :) invest in the future, not the past!

**Comment by Drakojin** [comment]
i am with you on that.

Sadly reality dictates mostly older technology :(
The paragraph for migration to junit5 is therefore of higher concern.
I doubt, that it is just done with switching to the vintage engine (in many cases at least)
